src/compiler/checker.ts

```diff
@@ -24433,6 +24428,10 @@ namespace ts {
             return getNamedMembers(propsByName);
         }
 
+        function typeHasCallOrConstructSignatures(type: Type): boolean {

```

Comment:
why do you need this in the checker API? the type object already has `getCallSignatures` and `getConstructSignatures` and `TypeChecker` already has `getSignaturesOfType`

Reply:
I wanted to reuse the code inside of `checker.ts`, but it would be wierd to have to pass itself into the helper function.